usr/lib/xtable/luaxt.c

```diff
@@ -0,0 +1,203 @@
+}
+
+#define LIBXTABLE_SAVER_CB(hook)				\
+static void luaxt_##hook##_save(const void *entry, const struct xt_entry_##hook *hook)  \
+{								\
+	luaxt_run(L, "save", "luaxt_"#hook, 0, 0);	\
+}
+
+#define LIBXTABLE_NEWCB(hook)  		\
+	LIBXTABLE_HELPER_CB(hook)   	\
+	LIBXTABLE_INITER_CB(hook)   	\
+	LIBXTABLE_PARSER_CB(hook)   	\
+	LIBXTABLE__FINALCHECKER_CB(hook)	\
+	LIBXTABLE_PRINTER_CB(hook)  	\
+	LIBXTABLE_SAVER_CB(hook)
+
+LIBXTABLE_NEWCB(match);
+LIBXTABLE_NEWCB(target);
+
+
+#define LIBXTABLE_NEWREG(hook)					\
+static struct xtables_##hook luaxt_##hook##_reg = {		\
+	.version = XTABLES_VERSION,				\
+	.name = LUAXTABLE_MODULE,					\
+	.size = XT_ALIGN(sizeof(luaxtable_info_t)),		\
+	.userspacesize = 0,					\
+	.help = luaxt_##hook##_help,			\
+	.init = luaxt_##hook##_init,			\
+	.parse = luaxt_##hook##_parse,		 	\
+	.final_check = luaxt_##hook##_finalcheck,	  	\
+	.print = luaxt_##hook##_print,		  	\
+	.save = luaxt_##hook##_save			 	\
+}
+
+LIBXTABLE_NEWREG(match);
+LIBXTABLE_NEWREG(target);
+
+static inline int luaxt_checkint(lua_State *L, int idx, const char *key)
+{
+	if (lua_getfield(L, idx, key) != LUA_TNUMBER) {
+		pr_err("invalid \'%s\' in match\n", key);
+		return 0;
+	}
+	int ret = lua_tointeger(L, -1);
+	lua_pop(L, 1);
+	return ret;
+}
+
+#define LIBXTABLE_LIB_CB(hook)				\
+static int luaxt_##hook(lua_State *L)		\
+{							\
+	luaL_checktype(L, 1, LUA_TTABLE);		\
+							\
+	luaxt_##hook##_reg.revision = luaxt_checkint(L, 1, "revision");		\
+	luaxt_##hook##_reg.family = luaxt_checkint(L, 1, "family");		\
+	xtables_register_##hook(&luaxt_##hook##_reg);		\
+	lua_pushvalue(L, 1);						\
+	lua_rawsetp(L, LUA_REGISTRYINDEX, "luaxt_"#hook);		\
+	return 0;							\
+}
+
+LIBXTABLE_LIB_CB(match);
+LIBXTABLE_LIB_CB(target);
+
+static const luaL_Reg luaxt_lib[] = {
+	{"match", luaxt_match},
+	{"target", luaxt_target},
+	{NULL, NULL}
+};
+
+static const luaxt_flags_t luaxt_family[] = {
+	{"UNSPEC", NFPROTO_UNSPEC},
+	{"INET", NFPROTO_INET},
+	{"IPV4", NFPROTO_IPV4},
+	{"IPV6", NFPROTO_IPV6},
+	{"ARP", NFPROTO_ARP},
+	{"NETDEV", NFPROTO_NETDEV},
+	{"BRIDGE", NFPROTO_BRIDGE},
+	{NULL, 0}
+};
+
+static int luaopen_luaxt(lua_State *L)
+{
+	luaxt_flags_t *flag;
+	luaL_newlib(L, luaxt_lib);
+	lua_newtable(L);
+	for (flag = luaxt_family; flag->name; flag++) {
+		lua_pushinteger(L, flag->value);
+		lua_setfield(L, -2, flag->name); /* namespace[name] = value */
+	}
+	lua_setfield(L, -2, "family"); /* lib.namespace = namespace */
+	return 1;
+}
+
+static int __attribute__((constructor)) _init(void)
+{
+	if ((L = luaL_newstate()) == NULL) {
+		pr_err("Failed to create Lua state\n");
+		return -ENOMEM;
+	}
+	luaL_openlibs(L);
+	luaL_requiref(L, "luaxt", luaopen_luaxt, 1);
+
+	if (luaL_dofile(L, "libxt_"LUAXTABLE_MODULE".lua") != LUA_OK) {
+		pr_err("Failed to load Lua script: %s\n", lua_tostring(L, -1));
+		return -ENOENT;
+	}
+	return 0;
+}
+
+static void __attribute__((destructor)) _fini(void)
+{
+	if (L) lua_close(L);
+}
```

Comment:
```suggestion
}

```

please mind this on every file